src/mds/objects.py

```diff
@@ -226,6 +226,161 @@ async def create_object_for_id(
     return JSONResponse(response, HTTP_201_CREATED)
 
 
+@mod.get("/objects")
+async def get_objects(
```

Comment:
If I understand it from the design doc linked in PXP-5529, this is sort of moving us towards the idea of rebranding MDS as a generalized "object management" service, is that right? (Just want to make sure I'm getting the context)

Reply:
Yeah that’s right. For the `GET /objects` endpoint in particular, note that Indexd records corresponding with metadata objects are returned in the response. I think the idea with the Object Management Service is to bring together various components of Gen3 (MDS, Fence, Indexd, SSJDispatcher, Indexs3client) to support a submission flow in which PFB files can be uploaded to the data lake (which consists of a single s3 bucket at the moment) by running a single gen3-client command.

During submission, both a Metadata object and an Indexd record pair are created and populated with info by an Indexs3client job, the Indexd record with calculated `hashes`, `size`, and `urls` info and the metadata object with `_bucket`, `_filename`, `_file_extension`, and `_upload_status`.

Reply:
Gotcha. That makes a lot of sense to me. Thanks!